common/fixture/atxs.go

```diff
@@ -41,42 +45,39 @@ func (g *AtxsGenerator) WithEpochs(start, n int) *AtxsGenerator {
 }
 
 // Next generates VerifiedActivationTx.
-func (g *AtxsGenerator) Next() *types.VerifiedActivationTx {
-	var atx types.VerifiedActivationTx
-
+func (g *AtxsGenerator) Next() *wire.ActivationTxV1 {
 	var prevAtxId types.ATXID
 	g.rng.Read(prevAtxId[:])
 	var posAtxId types.ATXID
 	g.rng.Read(posAtxId[:])
-	var nodeId types.NodeID
-	g.rng.Read(nodeId[:])
 
-	signer, err := signing.NewEdSigner()
+	signer, err := signing.NewEdSigner(signing.WithKeyFromRand(g.rng))
 	if err != nil {
 		panic("failed to create signer")
 	}
 
-	atx = types.VerifiedActivationTx{
-		ActivationTx: &types.ActivationTx{
-			InnerActivationTx: types.InnerActivationTx{
-				NIPostChallenge: types.NIPostChallenge{
-					Sequence:       g.rng.Uint64(),
-					PrevATXID:      prevAtxId,
-					PublishEpoch:   g.Epochs[g.rng.Intn(len(g.Epochs))],
-					PositioningATX: posAtxId,
-				},
-				Coinbase: wallet.Address(signer.PublicKey().Bytes()),
-				NumUnits: g.rng.Uint32(),
+	atx := &wire.ActivationTxV1{
+		InnerActivationTxV1: wire.InnerActivationTxV1{
+			NIPostChallengeV1: wire.NIPostChallengeV1{
+				Sequence:         g.rng.Uint64(),
+				PrevATXID:        prevAtxId,
+				PublishEpoch:     g.Epochs[g.rng.Intn(len(g.Epochs))],
+				PositioningATXID: posAtxId,
 			},
-			SmesherID: nodeId,
+			Coinbase: wallet.Address(signer.PublicKey().Bytes()),
+			NumUnits: g.rng.Uint32(),
 		},
 	}
+	atx.Sign(signer)
+	return atx
+}
 
-	atx.SetEffectiveNumUnits(atx.NumUnits)
-
-	var atxId types.ATXID
-	g.rng.Read(atxId[:])
-	atx.SetID(atxId)
-
-	return &atx
+func ToVerifiedAtx(t testing.TB, watx *wire.ActivationTxV1) *types.VerifiedActivationTx {
+	t.Helper()
+	atx := wire.ActivationTxFromWireV1(watx)
+	atx.SetEffectiveNumUnits(watx.NumUnits)
+	atx.SetReceived(time.Now())
+	vAtx, err := atx.Verify(0, 1)
+	require.NoError(t, err)
+	return vAtx
 }
```

Comment:
Ideally we get rid of the `VerifiedActivationTx` type and add its information (`BaseHeight` and `Ticks`) to `types.ActivationTx`.